src/scene/Scene.cpp

```diff
@@ -19,225 +18,208 @@ using namespace glm;
-    std::vector<Primitive *> nonPartPrimitives; // Primitives without ScenePart
-    ScenePart *_sp;  Primitive *_p;
-    for(size_t i = 0 ; i < s.primitives.size () ; i++){ // Fill parts
-        _p = s.primitives[i];
-        _sp = _p->part.get();
-        if(_sp != nullptr) parts.insert(_sp); // Primitive with part
-        else nonPartPrimitives.push_back(_p); // Primitive with no part
-    }
-    for(ScenePart *sp : parts){ // Handle primitives associated with ScenePart
-        std::shared_ptr<ScenePart> spc = std::make_shared<ScenePart>(*sp);
-        for(Primitive *p : spc->mPrimitives){
-            Primitive *_p = p->clone();
-            _p->part = spc;
-            this->primitives.push_back(_p);
-        }
-    }
-    for(Primitive *p : nonPartPrimitives){ // Handle primitives with no part
-        this->primitives.push_back(p->clone());
+Scene::Scene(Scene &s) {
+  if (s.bbox == nullptr)
+    this->bbox = nullptr;
+  else
+    this->bbox = std::shared_ptr<AABB>((AABB *)s.bbox->clone());
+  if (s.bbox_crs == nullptr)
+    this->bbox_crs = nullptr;
+  else
+    this->bbox_crs = std::shared_ptr<AABB>((AABB *)s.bbox_crs->clone());
+  std::set<ScenePart *> parts; // Pointer to each ScenePart, no repeats
+  std::vector<Primitive *> nonPartPrimitives; // Primitives without ScenePart
+  ScenePart *_sp;
+  Primitive *_p;
+  for (size_t i = 0; i < s.primitives.size(); i++) { // Fill parts
+    _p = s.primitives[i];
+    _sp = _p->part.get();
+    if (_sp != nullptr)
+      parts.insert(_sp); // Primitive with part
+    else
+      nonPartPrimitives.push_back(_p); // Primitive with no part
+  }
+  for (ScenePart *sp : parts) { // Handle primitives associated with ScenePart
+    std::shared_ptr<ScenePart> spc = std::make_shared<ScenePart>(*sp);
+    for (Primitive *p : spc->mPrimitives) {
+      Primitive *_p = p->clone();
+      _p->part = spc;
+      this->primitives.push_back(_p);
     }
+  }
+  for (Primitive *p : nonPartPrimitives) { // Handle primitives with no part
+    this->primitives.push_back(p->clone());
+  }
 
-    this->kdtree = shared_ptr<KDTreeNodeRoot>(
-        KDTreeNodeRoot::build(primitives)
-    );
+  this->kdtree = shared_ptr<KDTreeNodeRoot>(KDTreeNodeRoot::build(primitives));
 }
 
 // ***  M E T H O D S  *** //
 // *********************** //
 bool Scene::finalizeLoading() {
-	if (primitives.size() == 0) {
-		return false;
-	}
-
-    // #####   UPDATE PRIMITIVES ON FINISH LOADING   #####
-    UniformNoiseSource<double> uns(-1, 1);
-    for(Primitive *p: primitives){
-        p->onFinishLoading(uns);
+  if (primitives.size() == 0) {
+    return false;
+  }
+
+  // #####   UPDATE PRIMITIVES ON FINISH LOADING   #####
+  UniformNoiseSource<double> uns(-1, 1);
+  for (Primitive *p : primitives) {
+    p->onFinishLoading(uns);
+  }
+
+  // ################ BEGIN Shift primitives to originWaypoint
+  // ##################
+
+  // Translate scene coordinates to originWaypoint (to prevent wasting of
```

Comment:
what is "originWaypoint" referring to in this comment? the bbox min is moved to 0,0,0 in line 99ff.

Reply:
Well... I don't know. That part appears as a new change in the commit due to a reformat of the file that I did. The comment first appeared in in the first commit you did to this repository. f27b518, on 20/01/21@16.07h - "Init commit to github". I did not touch that part of the code at all.

Reply:
fair. Please delete the misleading comment :)